buildpacks/ruby/src/layers/bundle_install_configure_env_layer.rs

```diff
@@ -0,0 +1,94 @@
+use crate::RubyBuildpackError;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::generic::GenericMetadata;
+use std::path::Path;
+
+use crate::RubyBuildpack;
+use libcnb::build::BuildContext;
+use libcnb::layer::{Layer, LayerResult, LayerResultBuilder};
+
+use libcnb::layer_env::{LayerEnv, ModificationBehavior, Scope};
+
+/*
+
+# Set up environment for `bundle install`
+
+Must run before `execute_bundle_install_layer`. This is decoupled from dependency
+storage so that it can be updated without having to reinstall gems.
+
+## Layer dir:
+
+None
+
+## Set environment variables
+
+- `BUNDLE_CLEAN=1` - After successful `bundle install` bundler will automatically run `bundle clean`.
+- `BUNDLE_DEPLOYMENT=1` - Requires the `Gemfile.lock` to be in sync with the current `Gemfile`.
+- `BUNDLE_GEMFILE=<app-dir>/Gemfile` - Tells bundler where to find the `Gemfile`.
+- `BUNDLE_WITHOUT=development:test:$BUNDLE_WITHOUT` - Do not install `development` or `test` groups via bundle isntall. Additional groups can be specified via user config.
+
+## Environment NOT set by this buildpack
```

Comment:
I think thats reasonable. I moved some of the comments next to the env var they're modifying in the code. I added a link to the other module instead of re-explaining them. Working on that change now.